quinn-proto/src/connection/assembler.rs

```diff
@@ -64,23 +62,28 @@ impl Assembler {
             None => return false,
         };
 
+        let offset = match self.state {
+            State::Ordered { ref mut offset } => offset,
```

Comment:
It's the newer pattern, but I disagree that it's a better one, and I don't think there's a consensus either way (hence no clippy lint). I prefer this approach because the type of `offset` is determined by its immediate context, rather than the type of whatever was passed in to the `match`. This makes the type more obvious, especially for larger matches and matches on variables where it isn't immediately apparent whether they contain a reference.

Reply:
Hmm, ok.